src/Module/BaseApi.php

```diff
@@ -292,14 +293,54 @@ public static function checkThrottleLimit()
 		}
 	}
 
-	public static function getContactIDForSearchterm($searchterm)
+	public static function getContactIDForSearchterm(string $screen_name, int $cid, int $uid)
 	{
-		if (intval($searchterm) == 0) {
-			$cid = Contact::getIdForURL($searchterm, 0, false);
+		if (!empty($cid)) {
+			return $cid;
+		}
+
+		if (strpos($screen_name, '@') !== false) {
+			$cid = Contact::getIdForURL($screen_name, 0, false);
 		} else {
-			$cid = intval($searchterm);
+			$user = User::getByNickname($screen_name, ['uid']);
+			if (!empty($user['uid'])) {
+				$cid = Contact::getPublicIdByUserId($user['uid']);
+			}
+		}
+
+		if (empty($cid) && ($uid != 0)) {
+			$cid = Contact::getPublicIdByUserId($uid);
 		}
 
 		return $cid;
 	}
+
+	/**
+	 * Set values for RSS template
+	 *
+	 * @param array $arr Array to be passed to template
+	 * @param int   $cid Contact ID of template
+	 * @return array
+	 */
+	public static function addRSSValues(array $arr, int $cid)
```

Comment:
plz move this class as `private function addRSSValues(array arr, int $cid)` into `ApiRespone`